lib/components/result-view/display.js

```diff
@@ -1,27 +1,61 @@
 /* @flow */
 
 import React from "react";
+import { toJS } from "mobx";
 import {
   DisplayData,
   ExecuteResult,
   StreamText,
   KernelOutputError,
-  Output
+  Output,
+  Media,
+  RichMedia
 } from "@nteract/outputs";
-import { MediaComponents } from "./transforms";
+import PlotlyTransform from "@nteract/transform-plotly";
+import { VegaLite1, VegaLite2, Vega2, Vega3 } from "@nteract/transform-vega";
 
-type Props = {};
+type Props = { output: any };
 
-class Display extends React.PureComponent<Props> {
-  render() {
-    return (
-      <Output {...this.props}>
-        <ExecuteResult expanded>{MediaComponents()}</ExecuteResult>
-        <DisplayData expanded>{MediaComponents()}</DisplayData>
-        <StreamText expanded />
-        <KernelOutputError expanded />
-      </Output>
-    );
-  }
+// All supported media types for output go here
+export const supportedMediaTypes = (
+  <RichMedia>
+    <Vega3 />
+    <Vega2 />
+    <PlotlyTransform />
+    <VegaLite2 />
+    <VegaLite1 />
+    <Media.Json />
+    <Media.JavaScript />
+    <Media.HTML />
+    <Media.Markdown />
+    <Media.LaTeX />
+    <Media.SVG />
+    <Media.Image />
+    <Media.Plain />
+  </RichMedia>
+);
+
+export function isTextOutputOnly(data: Object) {
+  const supported = React.Children.map(
+    supportedMediaTypes.props.children,
+    mediaComponent => mediaComponent.props.mediaType
+  );
+  const bundleMediaTypes = [...Object.keys(data)].filter(mediaType =>
+    supported.includes(mediaType)
+  );
+
+  return bundleMediaTypes.length === 1 && bundleMediaTypes[0] === "text/plain"
+    ? true
+    : false;
+}
+
+export default function Display(props: Props) {
+  return (
+    <Output output={toJS(props.output)}>
+      <ExecuteResult expanded>{supportedMediaTypes}</ExecuteResult>
+      <DisplayData expanded>{supportedMediaTypes}</DisplayData>
+      <StreamText expanded />
+      <KernelOutputError expanded />
+    </Output>
+  );
```

Comment:
I am unsure of how exactly this works like a react component, but it works and nicely done